### DISCUSSION TITLE
How to disable emojis

### DISCUSSION THREAD

**Discussion by DeadKper** [wezterm/wezterm#5527] [Q&A] [answered]
### What Operating System(s) are you running on?

Windows

### WezTerm version

wezterm 20240203-110809-5046fc22

### Ask your question!

I think I'm dumb, I found how to do everything else that I wanted to do in the documentation, I found the property ```assume_emoji_presentation``` and tried
```lua
config.font = wezterm.font({
  family = "FiraCode Nerd Font",
  assume_emoji_presentation = false,
})
```
but it didn't work, and I also tried setting it like this 
```lua
config.assume_emoji_presentation = false
```
but that isn't valid, tried using font_with_fallback and using the default fonts and setting assume_emoji to false, also tried ignoring the emoji font, also searched how to disable a font, checked the discussions but there's only one about emoji rules, and tried searching for it in general and nothing, and I'm pretty sure the solution is simple, I'm just dumb

**Comment by wez** [comment]
What do you mean by disable emoji?

If you mean "how do I disable ligatures", which are the special characters formed by combining two different characters and replacing them with an alternative symbol, then https://wezfurlong.org/wezterm/config/font-shaping.html will tell you how to do that.

If you mean "I don't want to see smiley face emoji", then there is no way to turn those off, but you can choose to use a different emoji font by adding your preferred emoji font to your font fallback https://wezfurlong.org/wezterm/config/fonts.html#fallback

If you mean "when using unicode 14 or later, I don't want to use emoji presentation and would prefer to use the textual representation of glyphs" then you probably want to set https://wezfurlong.org/wezterm/config/lua/config/unicode_version.html accordingly and may also want to select an older, less capable, set of fonts that only have the older textual presentation flavor of emoji.   It is unlikely that you want this unless you are a unicode aficionado or have a very specific problem with unicode compatibility that you need to resolve.

**Comment by DeadKper** [comment]
I do want ligatures, that's the reason why I switched to wezterm from alacritty, I don't mind having emoji support but I want to have the text representation of all characters that can have it, like the icon when working in a lua folder will show an emoji instead of a textual moon glyph, I tried setting unicode_version to 8 and 9 but it still shows the emoji, maybe I need to restart wezterm but I'm quite busy so I'll wait to get home to try again, what I want is basically to be able to remove/disable a built-in font, this one to be precise: "Noto Color Emoji"

**Reply by wez** [reply]
You can't disable that font, but you can put your preferred alternative in the font fallback as I mentioned above.

**Comment by fennecfx** [comment, marked as answer]
@DeadKper Try setting `assume_emoji_presentation = true` when specifying the font family. This tells wezterm/harfbuzz that your font is capable of showing a colorful emoji and allow that font to render instead of the built-in emoji font (Noto Color Emoji)

**Reply by DeadKper** [reply]
Yep, I'm dumb, this is the answer, is not that I didn't try, I was spelling it like "assume_emoji_REpresentation" (obviously in minus), don't know why I didn't copy-paste and don't know why this time wezterm didn't scream at me telling me it was not valid config, only problem is that ligatures won't work correctly doing that, but there's a work around if you set it 2 times, the second time with emoji_presentation:
```lua
config.font = wezterm.font_with_fallback({
  {
    family = "FiraCode Nerd Font",
  },
  {
    family = "FiraCode Nerd Font",
    assume_emoji_presentation = true,
  },
})

```

**Comment by DeadKper** [comment]
Been checking a couple of things, the problem seems to be the unicode version (can't say I'm a unicode aficionado but I dislike emojis so I prefer textual representation) but I tried setting it from 6 to 13 and the lightning glyph still shows as an emoji while in alacritty it shows as text, I tried setting Noto Emoji (without color) to see if that helps but it still seems to prefer the colored version, I'm using the same font in alacritty and in wezterm but only in wezterm it shows as an emoji. in both cases I'm using the same font
Wezterm (unicode_version = 8 after restarting mux-server):
<img width="311" alt="Screenshot_3" src="https://github.com/wez/wezterm/assets/35360914/f8ed1f9f-bcbf-4848-a1b1-c116dc4b1ae5">
Alacritty (default config, just a theme change):
<img width="302" alt="Screenshot_4" src="https://github.com/wez/wezterm/assets/35360914/3bb9ed77-a02c-4631-bd32-8f89f9891689">

**Reply by fennecfx** [reply]
Give my suggestion a try. I found this discussion looking for a solution to the same issue and it worked for me. Remove all the  `unicode_version` and emoji font overrides/fallbacks.

In your case, try this:
```
config.font = wezterm.font({
  family = "FiraCode Nerd Font",
  assume_emoji_presentation = true,
})
```
